### DISCUSSION TITLE
UNet with 3D input and 2D output?

### DISCUSSION THREAD

**Discussion by dzenanz** [Project-MONAI/MONAI#7022] [Q&A] [answered]
I am creating a NN for segmenting ultrasound videos. Currently, I am using 3D approach, inputting the last 4 frames (512x512x4), getting the segmentation in the same shape, then discarding the first 3 frames, and only using the current frame. However, this feels wasteful.

Is it possible to output a single frame segmentation (512x512x1)? If yes, is it only the question of getting the `strides` parameter (and maybe `upsample_kernel_size`) to `DynUnet` correct, or is there more to it?

**Comment by KumoLiu** [comment, marked as answer]
Hi @dzenanz, you appear to be using 4 as a channel; perhaps you could make it a channel in the first dimension and then simply change the `output_channels` of the network.
Thanks!

**Reply by dzenanz** [reply]
Thank you, it works. It is retroactively obvious 😄
